### DISCUSSION TITLE
get bot voice channel on slash commands

### DISCUSSION THREAD

**Discussion by aurkaxi** [Rapptz/discord.py#8452] [Q&A] [answered]
I read your example basic_voice.py which takes `ctx.voice_client` works. but `interaction.voice_client.channel` doesn't work. maybe because app_commands doesn't care about ctx.
Also Documentation says abotu `VoiceClient` which says ` interaction doesn't have VoiceClient`
so how can i work around with this?
example code:
```py
@app_commands.command(
        name="join",
        description="Joins Voice Channel"
    )
async def join(self, interaction: discord.Interaction):
    if interaction.user.voice:
        print("you are in a voice")
        if interaction.VoiceClient.channel != interaction.user.voice.channel:
            print("but not where i am")
```
tried `interaction.client.voice.channel`
`interaction.guild.voice_client.channel`
`discord.VoiceClient.channel`
they don't have channel attribute.

**Comment by LostLuma** [comment]
The interaction has a `guild` attribute and every guild has a [`voice_client`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Guild.voice_client) attribute like Context.

**Reply by aurkaxi** [reply]
hmm that's a clever move but i still cannot retrieve `channel`

**Reply by aurkaxi** [reply]
voice_client returns `<discord.voice_client.VoiceClient object at 0x7f529fc88460>`

**Reply by LostLuma** [reply]
Can you explain what your first issue is exactly? According to your second comment everything is working as intended, the `VoiceClient` class just does not have a custom repr defined.

**Reply by aurkaxi** [reply]
i want to get the channel info, so that i can match it if An user and my bot is in the same channel or not.

**Reply by aurkaxi** [reply]
`interaction.user.voice.channel` returns the channel name and voice_client returns a huge code

**Reply by LostLuma** [reply]
I got that, I wanted to know what you tried so we can help you better.

You should be able to just test whether the `voice_client` attribute on the guild is not None and then access the channel attribute on it.

> voice_client returns a huge code

This is the string representation of the `VoiceClient` class that is shown when you print an object, and unlike other classes in the library it was not manually overwritten to show more detailed information. Since it is shown to you instead of `None` you can see that the voice client exists.

**Reply by aurkaxi** [reply]
didn't quite get it. voice_client doesn't have channel attribute. i can see if bot is connected or not. but i am having trouble how to know on which channel he is connected.  i tried  something like this
```py
        await interaction.user.voice.channel.connect()
        print(interaction.guild.voice_client.channel)
        print(interaction.user.voice.channel)
   ```
   
>Since it is shown to you instead of None you can see that the voice client exists. 

i know it exists, but in which channel...

**Reply by aurkaxi** [reply]
or rather you can help me fix this line
```py 
 if interaction.guild.voice_client.channel != interaction.user.voice.channel:
     print("you are not in the same voice channel as the bot")
 ```

**Reply by aurkaxi** [reply]
i don't want to know details about VoiceClient Class or The huge code. i just want the channel name.

**Reply by LostLuma** [reply]
There's nothing inherently wrong with this line, it works just fine for me. If you have an actual error feel free to post it.

**Reply by aurkaxi** [reply]
:) bruh, why can't i make you understand. i am not sharing a bug or error.
 i am just asking help that 'how can i know the NAME OF THE CHANNEL MY BOT IS CONNECTED TO`
 Just like i can know where the user is. How to know where the bot is?

**Reply by LostLuma** [reply]
You already print both the channel the bot is connected to and the channel the user is connected to to the console, all you need to do for the name is access the [`name`](https://discordpy.readthedocs.io/en/latest/api.html#discord.VoiceChannel.name) attribute.

> :) bruh, why can't i make you understand. i am not sharing a bug or error.

I apologize for trying to be helpful in understanding any issues you may be experiencing, sadly I can not read your thoughts so when you asked `or rather you can help me fix this line` I ran your code on a test bot and determined there to be no issues. Since you are too impatient to ask concise questions in a single comment and insist on asking multiple at once while getting mad I am not responding to everything this will be my last comment here, good luck figuring out the rest.

**Reply by aurkaxi** [reply]
>You already print both the channel the bot is connected to 

no i didn't, that prints `<discord.voice_client.VoiceClient object at 0x7f529fc88460>` which is obviously not the `.name`
i kept saying multiple time all i need is the name of the channel. and all you were asking if there was an error. 
Sadly we got some misunderstanding between us. I understand you are not my personal friend or employee so I obviously cannot get mad at, you are just another random helpful geek. I apologize for my behavior. 
And of course I will never be able to figure out the rest.

**Comment by aurkaxi** [comment, marked as answer]
I don't need answers

**Reply by pedrocaribe** [reply]
What you're looking for is this:

```if interaction.user.voice.channel.id == interaction.client.voice_clients[0].channel.id:```
